sdk/core/azure-core-test/src/main/java/com/azure/core/test/AzureTestWatcher.java

```diff
@@ -45,6 +55,10 @@ public void beforeTestExecution(ExtensionContext extensionContext) {
 
     @Override
     public void afterTestExecution(ExtensionContext context) {
+        if (!SHOULD_LOG_EXECUTION_STATUS.get()) {
+            return;
+        }
```

Comment:
Can we check the test status and always log if the test failed? The junit errors may not always give all the details.